src/aaudio/AudioStreamAAudio.cpp

```diff
@@ -981,4 +983,44 @@ Result AudioStreamAAudio::setOffloadEndOfStream() {
     return static_cast<Result>(mLibLoader->stream_setOffloadEndOfStream(stream));
 }
 
+void AudioStreamAAudio::updateDeviceIds() {
+    if (mLibLoader->stream_getDeviceIds == nullptr) {
+        return;
+    }
+    int deviceIdSize = 0;
+
+    // Query to get the size
+    {
+        std::vector<int32_t> tempDeviceIds; // Use a temporary vector
+        // Pass a nullptr as data on the first call
```

Comment:
Something weird and tricky happening here.
Please add an explanatory comment?

AAudio.h says "An array of size 16 should generally be large enough to fit all device identifiers.". 
Could you start with an array of 16 and only try again if there is no room?
(The downside of this is that the >16 code will not get tested much.)
I'm not sure what is the cleanest way to do this. But please consider alternatives and then thoroughly comment whatever you choose.

Reply:
See the new code. I made the assumption that 16 is enough and asked Oboe developers to file bugs